Cannot parse mut ref

Error when parsing mut reference

I tried this code:

```rust
enum Nat {
    S(Box<Nat>),
    Z,
}
fn test(x: &mut Nat) {
    let mut p = &mut *x;
    loop {
        match p {
            &mut Nat::Z => break,
            &mut Nat::S(ref mut n) => p = &mut *n,
        }
    }
}
```

I expected to see this happen: Parse correctly

Instead, this happened:

```
./test.rs:10:14: error: found unexpected token ‘mut’ in null denotation
   10 |             &mut Nat::S(ref mut n) => p = &mut *n,
      |              ^~~
./test.rs:10:18: error: failed to parse expr in match arm in match expr
   10 |             &mut Nat::S(ref mut n) => p = &mut *n,
      |                  ^~~
./test.rs:8:9: error: failed to parse statement or expression in block expression
    8 |         match p {
      |         ^~~~~
./test.rs:10:18: error: could not parse loop body in (infinite) loop expression
   10 |             &mut Nat::S(ref mut n) => p = &mut *n,
      |                  ^~~
./test.rs:7:5: error: failed to parse statement or expression in block expression
    7 |     loop {
      |     ^~~~
./test.rs:10:24: error: expecting ‘!’ but ‘(’ found
   10 |             &mut Nat::S(ref mut n) => p = &mut *n,
      |                        ^
./test.rs:10:24: error: failed to parse item in crate
```

### Meta

- c7728c8020334fecb47789163ba94326aab6208c

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -12082,7 +12082,7 @@ Parser<ManagedTokenSource>::parse_expr (int right_binding_power,
     {
       TokenId id = current_token->get_id ();
       if (id == SEMICOLON || id == RIGHT_PAREN || id == RIGHT_CURLY
-	  || id == RIGHT_SQUARE)
+	  || id == RIGHT_SQUARE || id == COMMA)
 	return nullptr;
     }
 

--- a/gcc/testsuite/rust/compile/match_break.rs
+++ b/gcc/testsuite/rust/compile/match_break.rs
@@ -1,0 +1,14 @@
+// { dg-additional-options "-frust-compile-until=ast" }
+enum Nat {
+    S(Box<Nat>),
+    Z,
+}
+fn test(x: &mut Nat) {
+    let mut p = &mut *x;
+    loop {
+        match p {
+            &mut Nat::Z => break,
+            &mut Nat::S(ref mut n) => p = &mut *n,
+        }
+    }
+}

